### PR TITLE
[RED-1964] Initial CBP work - Allow Before and After cursors to be specified. 

### PR DESCRIPTION
**Pull request by fbvilela**
### Description

This PR allows you to specify a `BeforeCursor` and `AfterCursor` to the `CursorPager` object, allowing you to make cursor based pagination requests and to loop through all the potential pages. 

### Example

```cs
var client = (ZendeskClient)serviceProvider.GetRequiredService<IZendeskClient>();

var cursorPager = new CursorPager { Size = 5 }; // forcing the pagination, default is 100.

var ticketCursorResponse = await client.Tickets.GetAllAsync(cursorPager);

while (ticketCursorResponse.Meta.HasMore)
{
    Console.WriteLine("Number of Tickets on this page: " + ticketCursorResponse.Count());
    foreach(var ticket in ticketCursorResponse)
    {
        Console.WriteLine("the id of this ticket is:" + ticket.Id);
    }

    cursorPager.AfterCursor = ticketCursorResponse.Meta.AfterCursor; // make sure the next request includes the after_cursor
    Console.WriteLine("fetching the next page");
    ticketCursorResponse = await client.Tickets.GetAllAsync(cursorPager);
}
```

### CODE REVIEW

File: ZendeskApiRepl/Program.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// See https://aka.ms/new-console-template for more information
```
Comment (mikerogers123) [inline ZendeskApiRepl/Program.cs:1]: I personally feel that the integration tests you have added serve as adequate documentation. Do you still see value in this?

**Comment by mikerogers123** [comment]
@fbvilela i think you need to increment the [version number](https://github.com/justeat/ZendeskApiClient/blob/master/src/ZendeskApi.Build/ZendeskApi.Commons.props)

I actually have an open PR to add some contributing guidance as it's not overly clear perhaps

File: ZendeskApiRepl/Program.cs (hunk 1 above)
Comment (fbvilela) [reply ZendeskApiRepl/Program.cs]: yeah good point, thanks! I have removed it and added the version number.

**Comment by fbvilela** [comment]
@mikerogers123 I have addressed your comment about the version number. Let me know if this is good to go 🙏

**Comment by fbvilela** [comment]
all done, @mikerogers123

**Review by mikerogers123 — APPROVED** [review]

**Pull request merged**